src/hooks/useBlockTimestamp.ts

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from 'react'
+
+import useAsync from './useAsync'
+
+import { useWeb3ReadOnly } from './wallets/web3'
+
+const INTERVAL = 1_000
```

Comment:
The block timestamp is in seconds and this hook is not necessarily recovery-specific.

Reply:
I understand but wherever we use this hook it would result in rerenders/side effects and possibly expensive computations every second. For the dashboard widget it is only necessary to be updated every 60s as far as I understand. We could pass in the interval as a prop to solve this instead of hard-coding it here.

Reply:
I've implemented your suggestion in f941c8b16710dc849bf54f0197360f9e02afc7e2.